### PR TITLE
ARTEMIS-1328 Improving direct delivery check

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -664,7 +664,7 @@ public void addTail(final MessageReference ref, final boolean direct) {
                if (intermediateMessageReferences.isEmpty() && messageReferences.isEmpty() && !pageIterator.hasNext() && !pageSubscription.isPaging()) {
                   // We must block on the executor to ensure any async deliveries have completed or we might get out of order
                   // deliveries
-                  if (flushExecutor() && flushDeliveriesInTransit()) {
+                  if (internalFlushExecutor(500, false) && flushDeliveriesInTransit()) {
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: Why 500 why not 100, if the issue was previously it was taking too long surely we want this as small as safely possible?

Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: I will change this to no wait at all.

**Comment by clebertsuconic** [comment]
Please.. don't merge this.. I'm changing this around.. will send another message when it's ok.

**Comment by clebertsuconic** [comment]
this is now ready...

@michaelandrepearce since you've looked.. mind looking again?

**Comment by michaelandrepearce** [comment]
@clebertsuconic only query i have 

was why this change?
-   public static final int CHECK_QUEUE_SIZE_PERIOD = 100;
+   public static final int CHECK_QUEUE_SIZE_PERIOD = 1000;

**Pull request closed without merging**